llama-index-core/llama_index/core/agent/runner/planner.py

```diff
@@ -341,73 +348,73 @@ def get_refine_plan_prompt_kwargs(
             "remaining_sub_tasks": remaining_sub_tasks_str.strip(),
         }
 
+    def _update_plan(self, plan_id: str, new_plan: Plan) -> None:
+        """Update the plan."""
+        # update state with new plan
+        self.state.plan_dict[plan_id] = new_plan
```

Comment:
I just realize that agent state doesn't seem to have any concurrent safety? Do we imagine that we'll need this in the future?

Reply:
I can't think of a specific reason why it might be needed right now 🤔 Two users should never be interacting the same task/plan

Reply:
Yeah I think I agree